framework/doc/content/source/functions/PiecewiseConstantFromCSV.md

```diff
@@ -0,0 +1,42 @@
+# PiecewiseConstantFromCSV
+
+!syntax description /Functions/PiecewiseConstantFromCSV
+
+## Description
+
+The `PiecewiseConstantFromCSV` function is used to load data from a CSV file into a function. The
+[PropertyReadFile.md] user object takes care of reading the CSV file, and the function queries
+information from it.
+
+It can assume CSV data
+
+- is sorted by element-id, in which case, when the function is evaluated at a point, it will locate the element containing it then return the value for that element in the CSV file
+- is sorted by node-id, in which case when the function is evaluated at a point, if this point coincides with a node, then it returns the value for that node in the CSV file. Else, it errors out.
+- is sorted by blocks, in which case, when the function is evaluated at a point, it will locate the element containing it then return the value for that element's block in the CSV file
+- defines an interpolation grid, in which case the function will locate the closest point in that interpolation grid, then return the value for that point in the CSV file
+
+
+For the latter case, the first columns of the CSV data must define the coordinates of each point forming the interpolation grid. The number of columns used to define these coordinates
```

Comment:
is it a grid though? if you're just finding the closest points it might as well be an unstructured point cloud

Reply:
changed. Didn't realize grid meant alignment, somehow